arangod/RocksDBEngine/RocksDBMetaCollection.cpp

```diff
@@ -869,17 +867,67 @@ bool RocksDBMetaCollection::haveBufferedOperations() const {
 }
 
 std::size_t RocksDBMetaCollection::revisionTreeDepth() const {
-  return 6;  // TODO make variable
+  std::size_t const treeCount = _revisionTree ? _revisionTree->count() : 0;
+  std::size_t bufferInserts = 0;
+  std::size_t bufferRemovals = 0;
+  {
+    std::unique_lock<std::mutex> guard(_revisionBufferLock);
+
+    rocksdb::SequenceNumber ignoreSeq = 0;
+    if (!_revisionTruncateBuffer.empty()) {
+      auto it = _revisionTruncateBuffer.rbegin();
+      if (it != _revisionTruncateBuffer.rend()) {
+        ignoreSeq = *it;
+      }
+    }
+
+    for (auto& pair : _revisionInsertBuffers) {
+      if (pair.first <= ignoreSeq) {
+        continue;
+      }
+
+      bufferInserts += pair.second.size();
+    }
+
+    for (auto& pair : _revisionRemovalBuffers) {
+      if (pair.first <= ignoreSeq) {
+        continue;
+      }
+
+      bufferRemovals += pair.second.size();
+    }
+  }
+
+  std::size_t const treeDepth = _revisionTree ? _revisionTree->maxDepth() : 3;
```

Comment:
this calculation is only used after the following `if` block, for the function's final return. So it could be moved below the `if` block for better readability locality.